conversion/conversion.go

```diff
@@ -242,6 +242,39 @@ func performSnapshotMigration(config writer.BatchWriterConfig, conv *internal.Co
 	return batchWriter
 }
 
+func performDataprocMigration(sourceProfile profiles.SourceProfile, targetProfile profiles.TargetProfile, config writer.BatchWriterConfig, conv *internal.Conv, client *sp.Client, infoSchema common.InfoSchema) (*writer.BatchWriter, error) {
+	common.SetRowStats(conv, infoSchema)
+	batchWriter := populateDataConv(conv, config, client)
+
+	//Adding all dataproc configs to a map
+	dataprocConfig := map[string]string{}
+	dataprocConfig["hostname"] = sourceProfile.Conn.Mysql.Host
+	dataprocConfig["port"] = sourceProfile.Conn.Mysql.Port
+	dataprocConfig["user"] = sourceProfile.Conn.Mysql.User
+	dataprocConfig["pwd"] = sourceProfile.Conn.Mysql.Pwd
+	dataprocConfig["targetdb"] = targetProfile.Dc.TargetDB
+	dataprocConfig["instance"] = targetProfile.Conn.Sp.Instance
+	dataprocConfig["project"] = targetProfile.Conn.Sp.Project
+	if len(targetProfile.Dc.Hostname) > 1 {
+		dataprocConfig["hostname"] = targetProfile.Dc.Hostname
+	}
+	if len(targetProfile.Dc.Port) > 1 {
+		dataprocConfig["port"] = targetProfile.Dc.Port
+	}
+	if len(targetProfile.Dc.Subnetwork) > 1 {
+		dataprocConfig["subnet"] = targetProfile.Dc.Subnetwork
```

Comment:
This seems odd. Why does `targetProfile` had `HostName` and other host related properties?

Reply:
Moved to `SourceProfile.SourceProfileConfig.ShardConfigurationDataproc` 

TODO (future): Rename `HostName` to a distinctive name to differentiate with `DirectConnectionConfig.Host`

Commit: https://github.com/cloudspannerecosystem/harbourbridge/pull/520/commits/d65e9ff652f5c9adf3145b8c9c860c02076a8ad6